src/sql/inspect.py

```diff
@@ -77,6 +77,51 @@ def _get_row_with_most_keys(rows):
     return list(rows[max_idx])
 
 
+def _is_numeric(value):
+    """Check if a column has numeric and not categorical datatype"""
```

Comment:
We should add test cases for these functions

Reply:
What if the column is categorical and has values like 1,2, ..etc. This function would still return True right?

Reply:
If a column only has numerical values then it is not a categorical column. 
if it contains values like 1,2,3 stored as a string then it is still a numeric column so `_is_numeric` will be true,
 but  the` _is_numeric_as_str` will give a message saying that there is a data mismatch

Reply:
added test `test_table_profile_data_mismatch` to check if a message and styling is applied to a column with data mismatch. `is_numeric` is an internal function being used in every column to determine the type of column it is. I think it is being verified in the existing tests and we don't need a separate test for it.

Reply:
Ok. This part is not very clear to me. 
As per my understanding, categorical columns can be either numeric or strings. So in one dataset `sex` can be represented as `Male` and `Female`, in some other dataset it can be represented as `1` and `2`.
Are we assuming the latter case will never happen?

We add testcase for internal functions as well if the function is significant. Especially in this case it'll help to understand the function better through different testcases.

Reply:
Sure, I'll add the testcase for the internal function. 

In the categorical column part, we have tried to replicate the pandas `description` function. So any time the categorical classes are represented in their encoded forms `eg. male and female represented as 0 1`, we treat them as numerical.  I think pandas does it because if a category has many classes which are in the encoded form, it will be tough to identify it as categorical. So we end up treating them as numerical columns.